File: backend/pawnscan/src/main/java/edu/cit/garbo/pawnscan/features/reports/dto/ReportResponse.java
package edu.cit.garbo.pawnscan.features.reports.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ReportResponse {

    private Long id;
    private String serialNumber;
    private String itemModel;
    private String description;
    private edu.cit.garbo.pawnscan.features.reports.entity.ReportStatus status;
    private LocalDateTime createdAt;
    private List<ReportFileResponse> files;
}

File: mobile/app/src/main/java/com/cit/pawnscan/features/reports/api/ReportResponse.kt
package com.cit.pawnscan.features.reports.api

data class ReportResponse(
    val id: Long?,
    val serialNumber: String?,
    val itemModel: String?,
    val description: String?,
    val status: String?,
    val rejectionReason: String?,
    val createdAt: String?,
    val files: List<ReportFileResponse>?
)
